data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessor.java

```diff
@@ -97,4 +101,30 @@ public boolean isReadyForShutdown() {
     @Override
     public void shutdown() {
     }
+
+    private void mergeValueToEvent(final Event recordEvent, final String key, final Object value) {
```

Comment:
Consider combining these with the use of Java 8 lambdas.

Here is some starting point for one way to do it.

```
private void mergeValueToEvent(final Event recordEvent, final String key, final Object value)
  mergeValue(recordEvent, () -> recordEvent.getMetadata().getAttribute(key), newValue -> recordEvent.put(key, newValue);
}

mergeValue(Supplier<Object> getter, Consumer<Object> setter) {
          final Object currentValue = getter.get();
        final List<Object> mergedValue = new ArrayList<>();
        if (currentValue instanceof List) {
            mergedValue.addAll((List<Object>) currentValue);
        } else {
            mergedValue.add(currentValue);
        }

        mergedValue.add(value);
        setter.accept(mergedValue);
}
```

Reply:
Thanks for the suggestion!